### PR TITLE
add ppc64le vector abi with tests

### PR DESCRIPTION
**Pull request by vtjnash**
also add conversion functions for VecElement to make it easier to construct

finishes fixing https://github.com/JuliaLang/julia/issues/16455

### CODE REVIEW

File: src/abi_ppc64le.cpp [hunk 1]
```diff
@@ -96,12 +127,37 @@ Type *preferred_llvm_type(jl_value_t *ty, bool isret)
...
+        }
+    }
+    // rewrite integer-sized (non-HFA) struct to an array
+    // the bitsize of the integer gives the desired alignment
+    if (size > 8) {
+        if (dt->alignment <= 8) {
+            return ArrayType::get(T_int64, (size + 7) / 8);
+        }
+        else {
+	    Type *T_int128 = Type::getIntNTy(jl_LLVMContext, 128);
```
Comment (tkelman) [inline src/abi_ppc64le.cpp]: tabs

File: src/codegen.cpp [hunk 2]
```diff
@@ -3709,7 +3709,6 @@ static Function *gen_cfun_wrapper(jl_function_t *ff, jl_value_t *jlrettype, jl_t
     // Create the call
     jl_cgval_t retval;
     if (lam == NULL) {
-        assert(nargs >= 0);
```
Comment (JeffBezanson) [inline src/codegen.cpp]: If these assertions are correct, might as well keep them.

Comment (tkelman) [inline src/codegen.cpp]: I thought the same thing but that var is unsigned?

**Comment by JeffBezanson** [comment]
LGTM.

File: src/codegen.cpp (hunk 2 above)
Comment (JeffBezanson) [inline src/codegen.cpp]: Ah, right.

File: test/ccall.jl [hunk 3]
```diff
@@ -712,4 +777,68 @@ elseif Sys.ARCH === :aarch64
...
+     ccall((:test_ppc64_vec2, libccalltest), V4xF32,
+    (Int64, V4xF32, V4xF32, V4xF32, V4xF32,
+     V4xF32, V4xF32, V4xF32, V4xF32, V4xF32,
+     V4xF32, V4xF32, V4xF32, V4xF32, V4xF32),
+    600000, (4, 3, 2, 1), (5, 4, 3, 2), (6, 5, 4, 3), (7, 6, 5, 4),
+    (14, 13, 12, 11), (15, 14, 13, 12), (16, 15, 14, 13), (17, 16, 15, 14), (18, 17, 16, 15),
+    (1024, 1023, 1022, 1021), (1025, 1024, 1023, 1022), (1026, 1025, 1024, 1023), (1027, 1026, 1025, 1024), (10028, 10027, 10026, 10025))
+
+else
+warn("ccall: no VecReg tests run for this platform")
```
Comment (tkelman) [inline test/ccall.jl]: do we really need this?

Comment (vtjnash) [inline test/ccall.jl]: we should be running VecReg tests on all supported platforms, or we are missing test coverage

Comment (tkelman) [inline test/ccall.jl]: ah. we're missing 32 bit

**Pull request merged**